tests/UnitTests/SparsityPattern/SparsityPatternTests.hpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <iomanip>
+
+#include <LinearAlgebra/SparsityPattern/Variable.hpp>
+#include <Utilities/Testing.hpp>
+#include <Utilities/TestHelpers.hpp>
+
+
+namespace GridKit
+{
+namespace Testing
+{
+    template<class ScalarT, typename IdxT>
+    class SparsityPatternTests
+    {
+    public:
+        SparsityPatternTests() = default;
+        ~SparsityPatternTests() = default;
+
+        /// Constructor, allocation, and initialization checks
+        TestOutcome variable()
+        {
+            TestStatus success = true;
+
+            const size_t n = 3;
+            std::vector<Sparse::Variable> x(n), p(n), f(n);
+            
+            // decide x, y, and z are variables
+            for (size_t i = 0; i < n; ++i)
+                x[i].setVariableNumber(i);
+            
+            // ...and sigma, rho, and beta are constant parameters
+            for (size_t i = 0; i < n; ++i)
+                p[i].setFixed(true);
+        
+            // initialize independent variables
+            x[0]() = 8.0; x[1]() = 20.0; x[2]() = 2.0/3.0;
+        
+            // set constant parameter values
+            p[0] = 10.0; p[1] = 8.0/3.0; p[2] = 28.0;
+            
+            // The residualFunction computes f
+            residualFunction(f, x, p);
+
+            // Check dependenices of f[0] (depends on x[0] and x[1])
+            {
+                const Sparse::Variable::DependencyMap& dependencies = 
+                    (f[0]).getDependencies();
+            
+                success *= (dependencies.size() == 2);
+                success *= (dependencies.find(0) != dependencies.end());
+                success *= (dependencies.find(1) != dependencies.end());
+                success *= (dependencies.find(2) == dependencies.end());    
+            }
+            
+            // Check dependencies of f[1] (depends on x[0], x[1], x[2])
```

Comment:
Should be f[2] in the comment.

Reply:
Fixed in [6d97154](https://github.com/ORNL/GridKit/pull/63/commits/6d971544a1a2aea56df7f52ff2af39faabfa6561).